app/controllers/api/v1/profiles_controller.rb

```diff
@@ -1,9 +1,25 @@
 class Api::V1::ProfilesController < Api::BaseController
+  before_action :set_user, only: [:show]
+
   def show
-    @user = User.find_by_slug!(params[:id])
     respond_to do |format|
       format.json { render json: @user }
       format.yaml { render yaml: @user }
     end
   end
+
+  private
+
+  def set_user
+    @user =
+      if params[:id]
+        User.find_by_slug!(params[:id])
+      else
+        authenticate_or_request_with_http_basic do |username, password|
+          if (user = User.authenticate(username.strip, password))
+            user.private_payload
```

Comment:
The `set_user` method seems to be rendering two different things. If the `id` param is supplied, the `User` object is rendered, but if no `id` param is provided and the user can successfully authenticate, then `user.private_payload` (a hash) is rendered.

I think these 2 bits of behaviour should be separated into separate actions. It would help separate each bit of functionality into its area and would help simplify the application code & tests.

Reply:
@colby-swandale Are you suggesting a separate `PrivateProfilesController` with its own show action or that we should have custom actions on `ProfilesController`, `show_public` and `show_private`?  I stuck them both into one to be more REST-compliant but I could see bucking the trend here.

Reply:
Having a current user endpoint is probably the eventual end goal we want, but I think that's part of a larger block of work that needs to be considered first.

Maybe we could render the private fields If the user being rendered is the same user authenticating?

Reply:
> Maybe we could render the private fields If the user being rendered is the same user authenticating?

Yes, that is the goal here, when no id is given the user that is authenticating is rendered the private fields of their own profile. We specifically do this when no id is given since the regular sign in process on console has no knowledge of the user id, so we don't want the user to need to know and supply their own id. 

I've updated the pr in https://github.com/rubygems/rubygems.org/pull/2922/commits/5ab97a8a038fe49d95e212cccb1d24bef6145139 to be more intention-revealing with the helper names

Reply:
Apologies, I meant to say that we render the same `User` object, but we include the private fields when it's rendered if the authenticating user is the same user being retrieved. That way, we're still rendering the same 'thing' in both scenarios.

Reply:
@colby-swandale I've updated this to set `hide_mfa` on the `User` so that `user.payload` will respond appropriately and we can render the same `User` object